feeluown/components/volume_button.py

```diff
@@ -0,0 +1,83 @@
+from PyQt5.QtCore import Qt, QPoint
+from PyQt5.QtWidgets import (
+    QSlider,
+    QPushButton,
+    QStyle,
+)
+
+
+class VolumeSlider(QSlider):
+
+    def __init__(self, parent=None, initial_value=100):
+        super().__init__(parent)
+
+        self.setOrientation(Qt.Horizontal)
+        self.setMinimum(0)
+        self.setMaximum(100)
+        self.setValue(initial_value)
+        self.setWindowFlags(Qt.Popup)
+        self.setAttribute(Qt.WA_TranslucentBackground)
+
+    def is_mute(self):
+        return self.value() <= 0
+
+    def showEvent(self, event):
+        parent = self.parent()
+        if parent:
+            pgeom = parent.geometry()
+            geom = self.geometry()
+            if self.orientation() == Qt.Horizontal:
+                x = pgeom.width() + 2
+                y = (pgeom.height() - geom.height())//2
+            else:
+                x = (pgeom.width() - geom.width())//2
+                y = geom.height() + 2
```

Comment:
> Maybe Vertical is more suitable? 

Yeah, I will add a vertical volume slider option. I think your former advise is right, I should only provide widgets in `feeluown.components` for the ui designer to choose instead of adding it directly.

> If the slider is horizontal oriented, the progress slider will be covered by the volume slider

I noticed it, but the volume slider's size could be set in `mac.qss` or other theme files, which is provided by the theme designers. Maybe we should set a suitable value in the default theme.